### PR TITLE
Fix/267 runs details refresh

### LINKED ISSUE #267: Runs details refresh
Screenshot 2020-08-03 at 12 26 04

When user clicks on Refresh with open details, details page should be refreshed along with refreshing data in grid data.
Datagrid data after click on refresh are refreshed, problem is with component.

### PR DESCRIPTION
**Pull request by senelesithole**
Implements #267 

**How to Test:** 
Within your workflow, create Shell jobs that point to your local bash script, allow the jobs to have some running time in between. While the first job is running keeping refreshing the run-detail screen.
1. job 1 running, job 2 in queue.
2. job 1 fails/succeeds, job two running.
3. both jobs with final status.

### CODE REVIEW

File: ui/src/app/components/runs/run-detail/run-detail.component.ts [hunk 1]
```diff
@@ -44,6 +45,15 @@ export class RunDetailComponent implements OnInit, OnDestroy {
       this.loading = state.detail.loading;
       this.jobInstances = state.detail.jobInstances;
     });
+    this.onRefresh();
+  }
+
+  onRefresh() {
+    this.refreshSubject.subscribe((response) => {
```
Comment (kevinwallimann) [inline ui/src/app/components/runs/run-detail/run-detail.component.ts]: I think you should also unsubscribe in `ngOnDestroy`

**Review by kevinwallimann — COMMENTED** [review]
Works, I like it

File: ui/src/app/components/runs/run-detail/run-detail.component.ts (hunk 1 above)
Comment (senelesithole) [reply ui/src/app/components/runs/run-detail/run-detail.component.ts]: Unfortunately unsubscribing on ngOnDestroy destructs the main runs component view. I will leave it as is for now.

File: ui/src/app/components/runs/run-detail/run-detail.component.ts [hunk 2]
```diff
@@ -13,13 +13,13 @@
  * limitations under the License.
  */
 
-import { Component, Input, OnDestroy, OnInit } from '@angular/core';
+import { Component, Input, OnDestroy, OnInit, OnChanges, SimpleChanges } from '@angular/core';
```
Comment (jozefbakus) [inline ui/src/app/components/runs/run-detail/run-detail.component.ts]: You can remove imports: OnChanges, SimpleChanges

File: ui/src/app/components/runs/run-detail/run-detail.component.ts [hunk 3]
```diff
@@ -28,6 +28,7 @@ import { Subscription } from 'rxjs';
 })
 export class RunDetailComponent implements OnInit, OnDestroy {
   @Input('dagRunId') dagRunId: number;
+  @Input() refreshSubject: Subject<boolean> = new Subject<boolean>();
```
Comment (jozefbakus) [inline ui/src/app/components/runs/run-detail/run-detail.component.ts:31]: Looks fine, but I noticed unnecessary backend calls, every time you open detail it creates new subscription, so it if you open and close details you will end up with 10 calls. To fix you can do this:
1. In onRefresh method have only `this.store.dispatch(new GetDagRunDetail(this.dagRunId));`
2. In ngOnInit `this.refreshSubscription = this.refreshSubject.subscribe((response) => {
        if (response) {
          this.onRefresh();
      }
    });` 
3. Do not forget unsubscribe from subscription `!!this.refreshSubscription && this.refreshSubscription.unsubscribe();`

**Review by jozefbakus — APPROVED** [review]

**Review by jozefbakus — CHANGES_REQUESTED** [review]
Sorry, I've accidentally approved it, but I posted some change requests

File: ui/src/app/components/runs/run-detail/run-detail.component.ts (hunk 2 above)
Comment (senelesithole) [reply ui/src/app/components/runs/run-detail/run-detail.component.ts]: 👍🏽

File: ui/src/app/components/runs/run-detail/run-detail.component.ts (hunk 3 above)
Comment (senelesithole) [reply ui/src/app/components/runs/run-detail/run-detail.component.ts:31]: changes have been made, thanks 😊  for this.

File: ui/src/app/components/runs/runs.component.html [hunk 4]
```diff
@@ -130,7 +130,7 @@
         <b>Status: </b>{{detail.status}}
       </clr-dg-detail-header>
       <clr-dg-detail-body>
-        <app-run-detail [dagRunId]="detail.id"></app-run-detail>
+        <app-run-detail [dagRunId]="detail.id" [refreshSubject]="refreshSubject.asObservable()"></app-run-detail>
```
Comment (jozefbakus) [inline ui/src/app/components/runs/runs.component.html]: Last thing, .asObservable() is unnecessary  ;)

**Review by jozefbakus — APPROVED** [review]

**Pull request merged**